test/framework/src/main/java/org/opensearch/node/MockNode.java

```diff
@@ -100,6 +100,21 @@ public MockNode(
         this(settings, classpathPlugins, null, forbidPrivateIndexSettings);
     }
 
+    public MockNode(
+        final Settings settings,
+        final Collection<Class<? extends Plugin>> classpathPlugins,
+        final Path configPath,
+        final boolean forbidPrivateIndexSettings,
+        final Map<Class<? extends Plugin>, Class<? extends Plugin>> extendedPlugins
```

Comment:
Why do we need `extendedPlugins` here if we already have `classpathPlugins` ? If we need to use PluginInfo here, let's do that instead

Reply:
Because extended plugins contains the relationships between the classpath plugins.

i.e. classpathPluginA extends classpathPluginB.

Reply:
It does in one of the constructors in this class. I chose to support multiple different constructors to continue to support use-cases such as [this](https://github.com/opensearch-project/OpenSearch/blob/main/server/src/test/java/org/opensearch/indices/IndicesServiceCloseTests.java#L157-L161). For the majority of tests using classpathPlugins they don't need to declare extensions. For test cases that would need to declare extensions, they can use the respective constructor that supports it.

Reply:
I would suggest to have 2 constructors, as we did for `Node`:
- one that accepts ` final Collection<Class<? extends Plugin>> classpathPlugins,`
- one that accepts ` final Collection<PluginInfo> classpathPlugins,`

Reply:
Got it, I will need some time to analyze the different usages of the 7 different constructors in this class. It looks like this class has evolved over time to support many different scenarios for testing.